src/utils/account-with-lockup.js

```diff
@@ -33,46 +33,52 @@ async function signAndSendTransaction(receiverId, actions) {
-            await this.wrappedAccount.functionCall(lockupAccountId, 'add_full_access_key', {
-                new_public_key: newKeyPair.publicKey.toString()
-            }, BASE_GAS.mul(new BN(2)))
-
-            const tmpKeyStore = new InMemoryKeyStore()
-            await tmpKeyStore.setKey(this.connection.networkId, lockupAccountId, newKeyPair)
-            const tmpConnection = new Connection(this.connection.networkId, this.connection.provider, new InMemorySigner(tmpKeyStore))
-            const lockupAccount = new Account(tmpConnection, lockupAccountId)
-            await lockupAccount.deleteAccount(this.accountId)
-        } else {
-            let liquidBalance = new BN(await this.wrappedAccount.viewFunction(lockupAccountId, 'get_liquid_owners_balance'))
-            if (!liquidBalance.gt(missingAmount)) {
-                await this.wrappedAccount.functionCall(lockupAccountId, 'refresh_staking_pool_balance', {}, BASE_GAS.mul(new BN(3)))
-                liquidBalance = new BN(await this.wrappedAccount.viewFunction(lockupAccountId, 'get_liquid_owners_balance'))
-                if (!liquidBalance.gt(missingAmount)) {
-                    throw new WalletError('Not enough tokens.', 'signAndSendTransactions.notEnoughTokens')
-                }
-            }
-
-            await this.wrappedAccount.functionCall(lockupAccountId, 'transfer', {
-                // NOTE: Move all the liquid tokens to minimize transactions in the long run
-                amount: liquidBalance.toString(),
-                receiver_id: this.wrappedAccount.accountId
-            }, BASE_GAS.mul(new BN(2)))
-        }
+async function deleteLockupAccountIfPossible(lockupAccountId) {
+    const lockedBalance = new BN(await this.wrappedAccount.viewFunction(lockupAccountId, 'get_locked_amount'))
+    const poolAccountId = await this.wrappedAccount.viewFunction(lockupAccountId, 'get_staking_pool_account_id')
+
+    if (!poolAccountId && lockedBalance.eq(new BN(0))) {
+        console.info('Destroying lockup account to claim remaining funds', lockupAccountId)
+        const newKeyPair = KeyPair.fromRandom('ed25519')
+        await this.wrappedAccount.functionCall(lockupAccountId, 'add_full_access_key', {
+            new_public_key: newKeyPair.publicKey.toString()
+        }, BASE_GAS.mul(new BN(2)))
+
+        const tmpKeyStore = new InMemoryKeyStore()
+        await tmpKeyStore.setKey(this.connection.networkId, lockupAccountId, newKeyPair)
+        const tmpConnection = new Connection(this.connection.networkId, this.connection.provider, new InMemorySigner(tmpKeyStore))
+        const lockupAccount = new Account(tmpConnection, lockupAccountId)
+        await lockupAccount.deleteAccount(this.accountId)
     }
+}
 
-    return await this.wrappedAccount.signAndSendTransaction.call(this, receiverId, actions);
+export async function transferAllFromLockup(missingAmount) {
+    const lockupAccountId = getLockupAccountId(this.accountId)
```

Comment:
@patrick you need to add this condition if you want it to work on testnet:
`f (process.env.REACT_APP_USE_TESTINGLOCKUP && this.accountId.length < 64) {
        lockupAccountId = `testinglockup.${this.accountId}`
    } else {
        lockupAccountId = getLockupAccountId(this.accountId)
    }`